doc/design/embargo-full.md

```diff
@@ -67,6 +67,16 @@ Admins will be at liberty to manually release embargoed dandisets that have expi
 
 ### TODO policy details
 
+
+## Zarr download
+The MVP embargo zarr download design involves a request to our API server for every file being downloaded from the zarr archive.
+Since zarr archives can contain hundreds of thousands of files, there would be a corresponding number of requests, which might degrade performance of the API server.
+If this becomes an issue, we could:
+* Spin off the zarr download functionality into a separate microservice that can be scaled separately, and would isolate any accidental DDOS effects.
+  This could be as simple as adding a new django app and a new Heroku dyno, or as complex as a Lambda@Edge+CloudFront service.
```

Comment:
well, you know my opinion - overhead from such per-file operations would be too large and will result not only in necessity to 'scale up' but also in a significant performance impairment on access to such zarr file.

But it is all speculation at this point since I have not done any benchmarking. Ideally we should compare performance (e.g. to even just download or perform some prototypical operation which would need some "sparse access" to zarr) accessing a prototypical zarr "file" between
- access on public bucket
- access on private bucket with a specific IAM
- access on private bucket through "mint a URL per file" (suggested here)

My wild uninformed bet is that a "full download" of zarr would be at least x10 less efficient in the latter than former case even in case of "minting service" to not be overloaded.

Reply:
i would very much say "mint a url per file" is not going to be efficient for any downstream application like neuroglancer or xarray-based operations which do their own lookup and caching based on awareness of a zarr dataset. i'm going to suggest that this is perhaps not an acceptable MVP, and just like accessing any other s3 blob we rely on s3 services for zarr.

Reply:
Perhaps we can discuss this one live, but for me the current position is: we need data. That is, for the MVP effort we may be able to try out the straightforward "proxy service" and see how it does. Brian Helba has already developed some intuition that some simple caching techniques could probably provide a speedup over the base case. But again... data needed.

Reply:
We used to have a sample zarr under s3://dandiarchive/largeuploads/blobs/150slab.zarr/ but then we cleaned it up.  @satra - do you still have it somewhere?  would be also a good test for `upload` functionality